test/services/subgroups/create_test.rb

```diff
@@ -0,0 +1,41 @@
+require_relative "../../test_helper"
+
+class Subgroups::CreateTest < ActiveSupport::TestCase
```

Comment:
leaving this comment here for lack of a better place, since it is the "biggest bracket" test we have for google api.

controversial claim: i think there is a missing top-level test for this feature. we 

as is, we only ever test that all of the service objects call each other correctly, which i completely agree is the correct level of testing for the unit-level. but nowhere do we provide a top-level test that forces us to stub responses from the actual API in order to verify that our service objects calling each other correctly produced correct behavior under given assumptions about the shape of the API.

perhaps more to the point: we are missing a crucial piece of documentation. the shape of the responses from the google api were very challenging to figure out. by providing stubs, you would (1) leave bread crumbs for other developers (and future you), to understand just what exactly an `authorizer` (for example) looks like, and (2) provide a snapshot of what the response looked like at a time that we had working code so that if ever downstream changes cause our API to break, we have somewhere to go look and compare the actual API response with the response that we expect and konw how to handle. without such stubs, we have no documentation of our contract with the google API, and a lot more uncertainty if our contract with it ever breaks.

for all those reasons, i'd feel a lot more confident about the maintainability of these service objects if we added API stubs to the `subgroup_creation` feature test and asserted that groups actually were created, etc...

it seems unrealistic to try to do that before monday, so maybe i can think up a way that would be appropriate to add in such tests as step 1 of future work to extend this code? perhaps you could leave me some breadcrumbs in the form of links to documentation on the enpoints that each of our service objects actually make network calls to, so i can try to figure out what our requests and their responses look like when we make such calls?

all the above in mind, i realize that like minds can differ on testing philosophies in such situations, but i've tried to outline the case for why i think there are sound reasons for adding a top-level test that acts as something like a halfway contract test. happy to hear other opinions, and certainly don't want to block merging because such (admittedly annoying to write) tests don't yet exist.

Reply:
Recap of our convo off github: 
- My rationale for not writing higher level tests of this code was that the API calls are being done through a gem, not us writing HTTP requests/handling responses ourselves. Therefore if we wrote a high level test (to say, stub the response), we would actually be testing the gem itself and not our own code. I propose that we lock the gem version in so that if the API changes, we will know its the gem that needs to change in someway and not that something else broke it.